File: common/src/main/java/juuxel/adorn/block/entity/TradingStation.java
package juuxel.adorn.block.entity;

import juuxel.adorn.trading.Trade;
import juuxel.adorn.util.InventoryComponent;
import net.minecraft.network.chat.Component;

public interface TradingStation {
    Component getOwnerName();
    Trade getTrade();
    InventoryComponent getStorage();

    static TradingStation createEmpty() {
        return new TradingStation() {
            @Override
            public Component getOwnerName() {
                return Component.empty();
            }

            @Override
            public Trade getTrade() {
                return Trade.empty();
            }

            @Override
            public InventoryComponent getStorage() {
                return new InventoryComponent(TradingStationBlockEntity.STORAGE_SIZE);
            }
        };
    }
}

File: common/src/main/kotlin/juuxel/adorn/block/entity/TradingStation.kt
package juuxel.adorn.block.entity

import juuxel.adorn.trading.Trade
import juuxel.adorn.util.InventoryComponent
import net.minecraft.text.Text

interface TradingStation {
    val ownerName: Text
    val trade: Trade
    val storage: InventoryComponent

    companion object {
        fun createEmpty(): TradingStation = object : TradingStation {
            override val ownerName: Text = Text.empty()
            override val trade = Trade.empty()
            override val storage = InventoryComponent(12)
        }
    }
}
